File: src/main/java/com/moovie/backend/domain/usecase/activities/GetFriendsActivities.java
package com.moovie.backend.domain.usecase.activities;

import com.moovie.backend.domain.repository.UserActivitiesRepository;
import com.moovie.backend.domain.model.UserActivityListing;
import org.springframework.stereotype.Service;

@Service
public class GetFriendsActivities {

    private final UserActivitiesRepository userActivitiesRepository;

    public GetFriendsActivities(UserActivitiesRepository userActivitiesRepository) {
        this.userActivitiesRepository = userActivitiesRepository;
    }

    public UserActivityListing invoke(int page) {
        return userActivitiesRepository.getFriendsActivities(page);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/activities/GetFriendsActivities.kt
package org.mobyle.domain.usecase.activities

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.repository.UserActivitiesRepository
import org.mobyle.model.UserActivityListing

class GetFriendsActivities(private val repository: UserActivitiesRepository) {
    operator fun invoke(page: Int): UserActivityListing = runBlocking {
        repository.getFriendsActivities(page)
    }
}
